samza-core/src/main/java/org/apache/samza/runtime/AbstractApplicationRunner.java

```diff
@@ -106,4 +112,25 @@ final ExecutionPlan getExecutionPlan(StreamApplication app) throws Exception {
   final StreamManager getStreamManager() {
     return streamManager;
   }
+
+  /**
+   * Write the execution plan JSON to a file
+   * @param planJson JSON representation of the plan
+   */
+  final void writePlanJsonFile(String planJson) {
```

Comment:
I'd still prefer not adding a new public configuration/environment variable for this. Is there one we can reuse? E.g., can we put it in the logs directory and make the path in the bin/plan.html relative?